dns/tokenizer.py

```diff
@@ -378,7 +392,30 @@ def get(self, want_leading=False, want_comment=False):
                             raise dns.exception.UnexpectedEnd
                         if not (c2.isdigit() and c3.isdigit()):
                             raise dns.exception.SyntaxError
-                        c = chr(int(c) * 100 + int(c2) * 10 + int(c3))
+
+                        decimal = int(c) * 100 + int(c2) * 10 + int(c3)
+                        b = bytes((decimal,))
+                        if len(b) > 1:  # horrible hack (is python2)
+                            b = chr(decimal)
+
+                        utfcheck += b
+
+                        try:
+                            c = utfcheck.decode('utf-8')
+                            break
+                        except UnicodeDecodeError as exc:
+                            # Incomplete data
+                            c = self._get_char()
+                            if c == '':
+                                raise dns.exception.UnexpectedEnd
+
+                            if c == '\\':
+                                # Next byte in UTF-8 sequence
+                                continue
+                            else:
+                                # Incomplete UTF-8 sequence
+                                # don't raise with original .decode()
+                                raise exc
```

Comment:
why not raise with original exception?

Reply:
I wanted to be explicit about the fact that the sequence is bad/incomplete and "hide" the original exception